### ISSUE TITLE
Error detecting environment

### ISSUE THREAD

**Issue by romulof** [flowtype/flow-for-vscode#123]
VS Code console is reporting the following error:

```
[2017-05-04T14:54:01.464Z] [ERROR] flow - Unknown return code from Flow: 127
[2017-05-04T14:54:01.464Z] [ERROR] flow - Flow failed: /Users/romulo/test/node_modules/.bin/flow status --json /Users/romulo/test/src/Login.jsx. Error: {"exitCode":127,"stderr":"env: node: No such file or directory\n","stdout":""}
[2017-05-04T14:54:01.465Z] [ERROR] flow - Invalid JSON result from flow status --json /Users/romulo/test/src/Login.jsx. JSON:
''.
```

I think the probable reason for not finding `node` is because I'm using fish shell with NVM, but I have also configured NVM in my `.bash_profile` as fallback.

Is there a way for me to debug whats happening?

**Comment by svipben** [comment]
Duplicate of https://github.com/flowtype/flow-for-vscode/issues/121

**Comment by andrewboot** [comment]
@svipben Nope, it's not the same. Even error codes and descriptions are different. I also have encountered this issue (same output) and can't use Flow Extension for 2 days now. @romulof Have you find a way to resolve it?

**Comment by romulof** [comment]
@andrewboot, this way you are not properly using NVM, you just made it use your standard node installation as default, same as not having it installed. In my case, all node installations are made through NVM.

The only solution that works right now is starting VSCode process from terminal, as it carries all envvars present there.

**Comment by idan** [comment]
See https://github.com/Microsoft/vscode/issues/21655.

The short version is that you're probably using a non-bash shell on OSX. When you launch VS Code from the icon, it doesn't know about any path customization you've done in your shell's config (.bash_config, fish.config, etc). VS Code tries to figure it out for you, but it seems to have issues doing so with fish (and maybe zsh too). 

There's a [pull open (#29864)](https://github.com/Microsoft/vscode/pull/29864) which will hopefully solve this. I'm figuring out how to try it out and will report back.
